compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -521,7 +525,28 @@ class InferenceState {
         break;
       }
       case Effect.Mutate: {
-        if (valueKind.kind === ValueKind.Context) {
+        if (
+          isRefValueType(place.identifier) ||
+          isUseRefType(place.identifier)
+        ) {
+          if (this.#env.config.validateRefAccessDuringRender) {
```

Comment:
Previously refs would have fallen through to the  case where the valueKind is mutable, and we would record it as a mutation. Now we always treat refs specially, and either ignore them (default) or record a RefMutation error when ref validation is on.

Ref validation still has a bunch of false positives that we have to fix before we can enable it more broadly. However, the fix here avoids the false negatives even when the validation is off.